zerver/views/user_groups.py

```diff
@@ -128,12 +136,19 @@ def edit_user_group(
         if setting_group_id_name not in request_settings_dict:  # nocoverage
             continue
 
-        if request_settings_dict[setting_group_id_name] is not None and request_settings_dict[
-            setting_group_id_name
-        ] != getattr(user_group, setting_group_id_name):
-            setting_value_group_id = request_settings_dict[setting_group_id_name]
+        if request_settings_dict[setting_group_id_name] is not None:
+            setting_value = request_settings_dict[setting_group_id_name]
+            if isinstance(setting_value, str):
+                current_value = getattr(user_group, setting_name)
+                if current_value.name == setting_value:
+                    continue
+            else:
+                current_value = getattr(user_group, setting_group_id_name)
+                if current_value == setting_value:
```

Comment:
This is just to avoid additional query to get `name` when we can directly compare using ID if the new setting value is an integer.